### PR TITLE
Provide an all-in-one package for easier usage

### PR DESCRIPTION
**Pull request by zhangthen**
### Ⅰ. Describe what this PR did

- 1、Add seata-all-in-one jar 

Merge all the following jar‘s code into `seata-all ` jar, so that users can get all seata's code by dependent on `seata-all ` only.

```
io.seata:seata-common 
io.seata:seata-core 
io.seata:seata-config-core 
io.seata:seata-config-apollo 
io.seata:seata-config-nacos 
io.seata:seata-config-zk 
io.seata:seata-config-consul 
io.seata:seata-discovery-core 
io.seata:seata-discovery-consul 
io.seata:seata-discovery-eureka 
io.seata:seata-discovery-nacos 
io.seata:seata-discovery-redis 
io.seata:seata-discovery-sofa 
io.seata:seata-discovery-zk 
io.seata:seata-discovery-etcd3 
io.seata:seata-dubbo 
io.seata:seata-dubbo-alibaba 
io.seata:seata-rm 
io.seata:seata-rm-datasource 
io.seata:seata-spring 
io.seata:seata-tcc 
io.seata:seata-tm 
```

- 2、add `seata-bom` jar

Move all the third-party jars to `seata-bom ` to facilitate the management of third-party jars.

### Ⅱ. Does this pull request fix one issue?

#914 

I had tested the  `seata-all` in seata-samples, which will release after this PR release.

### CODE REVIEW

File: bom/pom.xml [hunk 1]
```diff
@@ -0,0 +1,336 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+    <groupId>io.seata</groupId>
+    <artifactId>seata-bom</artifactId>
+    <version>0.6.0-SNAPSHOT</version>
```
Comment (xingfudeshi) [inline bom/pom.xml]: ${revision}

File: bom/pom.xml [hunk 2]
```diff
@@ -0,0 +1,336 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (xingfudeshi) [inline bom/pom.xml]: copyright

File: common/pom.xml [hunk 3]
```diff
@@ -38,7 +38,6 @@
         <dependency>
             <groupId>ch.qos.logback</groupId>
             <artifactId>logback-classic</artifactId>
-            <optional>true</optional>
```
Comment (xingfudeshi) [inline common/pom.xml]: why remove optional?

File: discovery/seata-discovery-etcd3/pom.xml [hunk 4]
```diff
@@ -49,12 +43,10 @@
         <dependency>
             <groupId>io.etcd</groupId>
             <artifactId>jetcd-launcher</artifactId>
-            <scope>test</scope>
```
Comment (xingfudeshi) [inline discovery/seata-discovery-etcd3/pom.xml]: jetcd-launcher is only used for testing.

File: discovery/seata-discovery-etcd3/pom.xml [hunk 5]
```diff
@@ -49,12 +43,10 @@
...
         </dependency>
         <dependency>
             <groupId>org.testcontainers</groupId>
             <artifactId>testcontainers</artifactId>
-            <scope>test</scope>
```
Comment (xingfudeshi) [inline discovery/seata-discovery-etcd3/pom.xml]: testcontainers is only used for testing.

File: bom/pom.xml (hunk 1 above)
Comment (zhangthen) [reply bom/pom.xml]: ‘seata-bom’  has no parent.

File: common/pom.xml (hunk 3 above)
Comment (zhangthen) [reply common/pom.xml]: It is added in the bom.xml.

File: discovery/seata-discovery-etcd3/pom.xml (hunk 4 above)
Comment (zhangthen) [reply discovery/seata-discovery-etcd3/pom.xml]: See the bom.xml

File: discovery/seata-discovery-etcd3/pom.xml (hunk 5 above)
Comment (zhangthen) [reply discovery/seata-discovery-etcd3/pom.xml]: See the bom.xml

File: common/pom.xml (hunk 3 above)
Comment (xingfudeshi) [reply common/pom.xml]: > It is added in the bom.xml.

ok

File: discovery/seata-discovery-etcd3/pom.xml (hunk 4 above)
Comment (xingfudeshi) [reply discovery/seata-discovery-etcd3/pom.xml]: > See the bom.xml

got it.

File: discovery/seata-discovery-etcd3/pom.xml (hunk 5 above)
Comment (xingfudeshi) [reply discovery/seata-discovery-etcd3/pom.xml]: > See the bom.xml

got it

File: all/pom.xml [hunk 6]
```diff
@@ -0,0 +1,612 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (lovepoem) [inline all/pom.xml]: License header is reqiired

File: bom/pom.xml (hunk 2 above)
Comment (lovepoem) [inline bom/pom.xml]: License header is required

File: pom.xml [hunk 7]
```diff
@@ -143,6 +121,12 @@
             <version>${junit-jupiter.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.junit.platform</groupId>
+            <artifactId>junit-platform-launcher</artifactId>
+            <version>1.4.2</version>
```
Comment (lovepoem) [inline pom.xml]: Please extract the version number

**Comment by zhangthen** [comment]
@xingfudeshi @lovepoem  done.

**Comment by lovepoem** [comment]
lgtm

**Comment by slievrly** [comment]
@zhangthen  @ujjboy how to ensure the version consistency of the dependencies in bom.xml and all/pom.xml?

**Comment by lovepoem** [comment]
We can then discuss here  https://github.com/seata/seata/issues/1031

**Pull request merged**